v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+                current_user_data_field += sizeof(uint8_t);
+                break;
+            case TlgInINT8:
+                parsed_events[current_parsed_event_count].properties[property_index].int8_t_value = *(int8_t*)current_user_data_field;
+                current_user_data_field += sizeof(int8_t);
+                break;
+            case TlgInUINT16:
+                parsed_events[current_parsed_event_count].properties[property_index].uint16_t_value = *(uint16_t*)current_user_data_field;
+                current_user_data_field += sizeof(uint16_t);
+                break;
+            case TlgInINT16:
+                parsed_events[current_parsed_event_count].properties[property_index].int16_t_value = *(int16_t*)current_user_data_field;
+                current_user_data_field += sizeof(int16_t);
+                break;
+            case TlgInUINT32:
+                parsed_events[current_parsed_event_count].properties[property_index].uint32_t_value = *(uint32_t*)current_user_data_field;
+                current_user_data_field += sizeof(uint32_t);
+                break;
+            case TlgInINT32:
+                parsed_events[current_parsed_event_count].properties[property_index].int32_t_value = *(int32_t*)current_user_data_field;
+                current_user_data_field += sizeof(int32_t);
+                break;
+            case TlgInUINT64:
+                parsed_events[current_parsed_event_count].properties[property_index].uint64_t_value = *(uint64_t*)current_user_data_field;
+                current_user_data_field += sizeof(uint64_t);
+                break;
+            case TlgInINT64:
+                parsed_events[current_parsed_event_count].properties[property_index].int64_t_value = *(int64_t*)current_user_data_field;
+                current_user_data_field += sizeof(int64_t);
+                break;
+            }
+
+            property_index++;
+        }
+
+        parsed_events[current_parsed_event_count].property_count = property_index;
+
+        (void)InterlockedIncrement(&parsed_event_count);
+        WakeByAddressAll((void*)&parsed_event_count);
+    }
+}
+
+static char test_path[MAX_PATH];
+
+static void generate_trace_session(void)
+{
+    UUID uuid;
+    POOR_MANS_ASSERT(UuidCreate(&uuid) == RPC_S_OK);
+    int snprintf_result = snprintf(trace_session_name, sizeof(trace_session_name), "%s-%08X-%04X-%04X-%02X%02X-%02X%02X%02X%02X%02X%02X",
+        __FUNCTION__, uuid.Data1, uuid.Data2, uuid.Data3, uuid.Data4[0], uuid.Data4[1], uuid.Data4[2], uuid.Data4[3], uuid.Data4[4], uuid.Data4[5], uuid.Data4[6], uuid.Data4[7]);
+    POOR_MANS_ASSERT((snprintf_result > 0) && (snprintf_result < sizeof(trace_session_name)));
+}
+
+static TRACEHANDLE start_trace(uint8_t trace_level)
+{
+    // setup the etw consumer
+    TRACEHANDLE trace_session_handle;
+
+    (void)printf("Starting trace session %s\r\n", trace_session_name);
```

Comment:
>trace_session_name [](http://example.com/codeflow?start=50&length=18)

I did not follow the complete code when I am making this comment, but who guarantees that at this moment "trace_session_name" is even `'\0'` terminated? I don't see any ` = { 0 }` where `trace_session_name` is defined.

Reply:
Sort of "consider initializing all static globals" here and everywhere.

Reply:
We have a POOR_MANS_ASSERT with abort in it and we assert on the return of snprintf.
That should be enough per C11

Thus, the null-terminated output has been
completely written if and only if the returned value is nonnegative and less than n.